client/java/src/main/java/io/openlineage/client/transports/CustomTransportResolver.java

```diff
@@ -0,0 +1,42 @@
+/*
+/* Copyright 2018-2023 contributors to the OpenLineage project
+/* SPDX-License-Identifier: Apache-2.0
+*/
+
+package io.openlineage.client.transports;
+
+import java.util.Optional;
+import java.util.ServiceLoader;
+import java.util.function.Predicate;
+import java.util.stream.StreamSupport;
+
+public class CustomTransportResolver {
```

Comment:
Just thinking loudly: wouldn't it make sense to have `TransportBuilder` (instead of `CustomTransportBuilder`) and load all transport types using it?

Reply:
I initially thought it would be better to keep the original code without a change, although making everything work the same way makes sense. If we'll do that then I will need to add a dedicated builder for each existing transport (but it will replace the switch-case blocks and make the code cleaner). So if you think it makes sense to make this change, I'll go for it :)

Reply:
I think it makes sense because it's cleaner to build all the transports the same way. It can also save us writing extra test in ArgumentParserTest as `TransportConfigTypeIdResolver` will be already tested. 

@nataliezeller1 Please let me know if I am asking for too much refactoring for this PR. Otherwise I think it's worth making it consistent.

Reply:
@pawel-big-lebowski I've changed the existing transports to be loaded the same way, let me know if you have any comment